### PR TITLE
Add openOnLeave prop to Tooltip component

### PR DESCRIPTION
**Pull request by Kyubinhan**
## Description
This flag decides whether the popover should stay open after leaving the content.

## Notes
* This change is part of [this](https://riipen.atlassian.net/browse/PLT-7476) ticket.
* Add a blacklist to the build API script to ignore some of the files.

### CODE REVIEW

**Comment by daelynj** [comment]
Is this part of a ticket or related to one?

File: packages/riipen-ui/src/components/Tooltip.jsx [hunk 1]
```diff
@@ -78,6 +78,11 @@ class Tooltip extends React.Component {
      */
     open: PropTypes.bool,
 
+    /**
+     * Whether the popover should stay open after leaving the content
+     */
```
Comment (bolinkd) [inline packages/riipen-ui/src/components/Tooltip.jsx:83]:
```suggestion
    /**
     * Whether the popover should stay open after leaving the content.
     */
```

The documentation is generated using an npm script and we like to have a period at the end to clean up the docs.

File: packages/riipen-ui/src/components/Tooltip.jsx [hunk 2]
```diff
@@ -78,6 +78,11 @@ class Tooltip extends React.Component {
...
+    openOnLeave: PropTypes.bool,
```
Comment (bolinkd) [inline packages/riipen-ui/src/components/Tooltip.jsx:84]: I would aim for something a bit more descriptive to a boolean here
`keepOpen` or something similar.

openOnLeave implies to me that the tooltip will open if I do something

**Review by bolinkd — CHANGES_REQUESTED** [review]
We should also re-run the documentation generator 
in the packages/riipen-ui-docs run 
`npm run build && npm run format`

**Comment by daelynj** [comment]
There's a section in the documentation for "Tooltip Visibility" where it would be helpful to add an example of this new functionality: https://ui.riipen.com/components/tooltip/

**Comment by Kyubinhan** [comment]
> Is this part of a ticket or related to one?

Yes, this is the ticket https://riipen.atlassian.net/browse/PLT-7476, sorry.

**Comment by daelynj** [comment]
I'm also curious if it's possible to close the tooltip ever if `openOnLeave` is true? Should it be possible to close it?

**Comment by daelynj** [comment]
> > Is this part of a ticket or related to one?
> 
> Yes, this is the ticket https://riipen.atlassian.net/browse/PLT-7476, sorry.

No problem just want to understand why this change is necessary.

Is this PR for solving the main problem: Tooltips are not closing when clicking on the "X" icon
Or is this fixing a sub-problem that Maribel was mentioning in the ticket comments?

File: packages/riipen-ui/src/components/Tooltip.jsx (hunk 2 above)
Comment (Kyubinhan) [reply packages/riipen-ui/src/components/Tooltip.jsx:84]: Okay how about `keepOpenOnLeave` then? too long?

Comment (daelynj) [reply packages/riipen-ui/src/components/Tooltip.jsx:84]: I assume the idea is that once you hover, the tooltip stays open indefinitely until closed? (I assume there's a way to close it)

**Comment by Kyubinhan** [comment]
@daelynj 
The tooltip still can be closed through user's mouse event (click or scroll). and yes it will stay open until the user action.
It is related to the sub-problem which is making one of the tooltips stay open until the user closes it explicitly. Previously it was done like [this](https://github.com/riipen/platform-web/blob/master/src/components/InteractiveTooltip/index.jsx#L241) and I thought maybe it's better to have an explicit flag for staying open.

**Comment by Kyubinhan** [comment]
@bolinkd documentations have been generated and rename the prop name to `keepOpenOnMouseLeave`. Tooltip Visibility docs have been updated with the new prop as well.

**Comment by daelynj** [comment]
Wanna throw a couple screenshots of the documentation changes in here?

**Comment by Kyubinhan** [comment]
@daelynj  Like these?
image
image

**Comment by daelynj** [comment]
@Kyubinhan Cool yeah just wanted to see the actual UI there with the checkbox etc.

**Review by daelynj — APPROVED** [review]
Nice work

**Review by bolinkd — APPROVED** [review]
Turns out its been a bit since we ran the documentation 🙈 

Good work ✅

**Comment by bolinkd** [comment]
if you need assistance deploying the package, let me know

**Comment by Kyubinhan** [comment]
> if you need assistance deploying the package, let me know

Oh haha thx! Jordan showed me how to do it the last time :)

**Pull request merged**